File: src/test/java/com/example/scaffold/PingController.java
package com.example.scaffold;

import com.example.frameworks.domain.core.Payload;
import io.swagger.v3.oas.annotations.tags.Tag;
import org.springframework.web.bind.annotation.*;

import static com.example.frameworks.domain.core.StringFieldType.asString;

@Tag(name = "Restful controller连通性测试接口")
@RestController
public class PingController {
    @PostMapping("/ping")
    public String ping(Payload payload) {
        return payload.get("data", asString());
    }
}

File: src/test/kotlin/com/example/scaffold/PingController.kt
package com.example.scaffold

import com.example.scaffold.frameworks.valueobjects.payload.Payload
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RestController
import javax.annotation.Resource

@RestController
class PingController(@Resource private val pingUseCase: PingUseCase) {
    @PostMapping("/ping")
    fun ping(payload: Payload): Map<String, String> {
        val map = hashMapOf<String, String>();
        map["data"] = pingUseCase.execute(payload)
        return map;
    }
}
